test/maintest.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const greet = require(__dirname + '/../index');
+const assert = require('assert');
+
+describe('testing module index', function() {
+  describe('testing main function', function() {
+    it('should return inputed name', function() {
+      process.argv = ['year', 'andrew', 'andrew'];
```

Comment:
You only need to modify the process.argv[2] element, not the entire array, additionally, please keep in mind that since you're editing a variable which could be used by other functions in tests theoretically, it is preferable to save a copy of the unmodified version, modify the variable, run the test, and then reset the variable to its original value.
For example:
var orig = process.argv;
process.argv[2] = 'w/e';
tests...
process.argv = orig;
other stuff...